clang/lib/CodeGen/CodeGenTypes.h

```diff
@@ -229,6 +229,12 @@ class CodeGenTypes {
   const CGFunctionInfo &arrangeBuiltinFunctionCall(QualType resultType,
                                                    const CallArgList &args);
 
+  /// A SYCL device kernel function is a free standing function with
+  /// spir_kernel calling convention
```

Comment:
```suggestion
  /// spir_kernel calling convention.
```

Reply:
```suggestion
  /// target specific calling convention (e.g. spir_kernel for the SPIR target).
```
Technically we use OpenCL language calling convention, but I'm not sure if we should specify that in the comments. Typically, each offload target has its own calling convention. For example, OpenCL compiler emits `amdgpu_kernel` calling convention for AMD GPU targets and `ptx_kernel` for NVIDIA GPU targets.

NOTE: Setting calling convention might be not enough for decorating kernel functions properly. E.g. NVPTX target uses named metadata to identify kernel functions. 
https://docs.nvidia.com/cuda/nvvm-ir-spec/index.html#representation-of-properties
I suppose this means that SYCL compiler for NVPTX target must emit named metadata referring to the kernel in addition to using `ptx_kernel` calling convention.
Some SPIR tools consuming LLVM IR doesn't require additional metadata, so setting calling convention is enough. I don't know how NVPTX backend will handle LLVM w/o the metadata.
@npmiller, FYI.

Reply:
Thank you for the comment correction; I'll fix that. This change does produce the calling convention tags you noted (see the IR for the `kernel-caller-entry-point.cpp` test below). The changes to emit the kernel metadata will be included in the next PR; I deferred it from this one to ease code review.

Reply:
I fixed the comment, thank you!

Thank you for mentioning the kernel metadata and linking those previous PRs. As I mentioned, generation of kernel metadata is planned for the next PR I'll submit. Should it continue to emit the `kernel` annotation for NVPTX targets to align with the NVVM IR Specification? The code that we have in development currently does not do so.

Reply:
Answering my own question here. https://github.com/llvm/llvm-project/pull/120806 removed all the calls that caused the `kernel` NVVM metadata to be emitted as illustrated by both code and test changes. I checked for other calls in both LLVM/Clang and in the oneAPI sources and didn't find any. Resolving this comment now.